nix/loadtest.nix

```
@@ -0,0 +1,99 @@
+{ postgrest
+, envsubst
+, locust
+, openresty
+, postgresql
+, withTmpDb
+, writeShellScript
+, writeShellScriptBin
+, writeText
+}:
+let
+  runner =
+    writeShellScript "postgrest-loadtest-runner"
+      ''
+        set -euo pipefail
+
+        tmpdir=$(mktemp -d)
+        trap "rm -r $tmpdir" exit
+
+        export POSTGREST_TEST_DIR="$tmpdir"
+
+        ${postgrest}/bin/postgrest ${postgrestConf} &
+        postgrestPID=$!
+
+        export POSTGREST_API_URL="http://localhost:3000"
+
+        mkdir -p "$tmpdir"/{logs,conf}
+        touch "$tmpdir"/logs/{error.log,access.log}
+        ${envsubst}/bin/envsubst -i ${nginxConf} \
+          -o "$tmpdir/conf/nginx.conf"
+
+        ${openresty}/bin/openresty -p "$tmpdir" &
+        nginxPID=$!
+
+        cleanup() {
+          kill $postgrestPID $nginxPID || true
+          rm -r "$tmpdir"
+        }
+
+        trap cleanup exit
+
+        ${locust}/bin/locust -f ${../test/loadtest/locustfile.py} \
+            -H http://localhost:3080 --users 100 --spawn-rate 20 \
+            --headless -t 60s
+      '';
+
+  postgrestConf =
+    writeText "postgrest.conf"
+      ''
+        db-uri = "$(POSTGREST_TEST_CONNECTION)"
+        db-schema = "test"
+        db-anon-role = "postgrest_test_anonymous"
+        db-pool = 1
+        server-port = 3000
+        jwt-secret = "reallyreallyreallyreallyverysafe"
+      '';
+
+  nginxConf =
+    writeText "nginx.conf"
+      ''
+        daemon off;
+        error_log stderr info;
+        pid nginx.pid;
+
+        events {}
+
+        http {
+            access_log access.log;
```

I think you may be able to get more req/s if you enable buffering for the logs:

```suggestion
            access_log access.log  main buffer=32k flush=5m;
```